lib/scan_result.dart

```diff
@@ -14,49 +14,66 @@ abstract class _ScanResultMetadata {
 
 /// A scan result emitted by the scanning operation, containing [Peripheral] and [AdvertisementData].
 class ScanResult {
-  Peripheral peripheral;
+  final Peripheral peripheral;
 
   /// Signal strength of the peripheral in dBm.
-  int rssi;
+  final int rssi;
 
   /// An indicator whether the peripheral is connectable (iOS only).
-  bool isConnectable;
+  late final bool isConnectable;
 
   /// A list of UUIDs found in the overflow area of the advertisement data (iOS only).
-  List<String> overflowServiceUuids;
+  late final List<String> overflowServiceUuids;
```

Comment:
Late is there by mistake, i needed it at first but after refactoring the constructor into an init list it's not needed anymore, thanks to good catch! I'm going to look for all of these unnecessary late qualifiers.
iOS only:
Or it could be understood as default as an empty list on android, therefore not needing to branch the code for iOS/android anymore for the user. 
In other words, does being null carry any useful information for the user here? 
What difference does it make if it's null or it's an empty list of overflowIds semantically? Let's make the caller's life as easy as possible.
On the other hand, if there is a single use-case that the caller would branch on a null differently to an empty list it absolutely must be nullable!

Reply:
Null avoids the issues like `it's always empty` lacking the crucial detail - tested on Android. ;) Though it makes sense to put ease of use over potential unnecessary questions, let's keep it this way.